pkg/pbq/pbqmanager.go

```diff
@@ -0,0 +1,134 @@
+package pbq
+
+import (
+	"context"
+	"errors"
+	dfv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	"github.com/numaproj/numaflow/pkg/pbq/store"
+	"github.com/numaproj/numaflow/pkg/pbq/store/memory"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+	"go.uber.org/zap"
+	"sync"
+)
+
+var NonExistentPBQError error = errors.New("missing PBQ for the partition")
+
+// Manager helps in managing the lifecycle of PBQ instances
+type Manager struct {
+	options *store.Options
+	pbqMap  map[string]*PBQ
+	log     *zap.SugaredLogger
+}
+
+var createManagerOnce sync.Once
+var pbqManager *Manager
+var createManagerError error
+
+// NewManager returns new instance of manager
+// We don't intend this to be called by multiple routines.
+func NewManager(ctx context.Context, opts ...store.SetOption) (*Manager, error) {
+	options := store.DefaultOptions()
+	createManagerOnce.Do(func() {
```

Comment:
try to avoid `sync.Once` since we are ending up with global variables. Do we really need `sync.Once` ?

Reply:
its a defensive step as of now.